mailbox/plugin/spamassassin/src/main/java/org/apache/james/mailbox/spamassassin/SpamAssassinListener.java

```diff
@@ -58,25 +64,30 @@ public ExecutionMode getExecutionMode() {
 
     @Override
     public void event(Event event) {
-        LOGGER.debug("Event {} received in listener.", event);
-        if (event instanceof EventFactory.AddedImpl) {
-            EventFactory.AddedImpl addedToMailboxEvent = (EventFactory.AddedImpl) event;
-            if (isEventOnSpamMailbox(addedToMailboxEvent)) {
+        if (event instanceof MessageMoveEvent) {
+            MessageMoveEvent messageMoveEvent = (MessageMoveEvent) event;
+            if (isMessageMovedToSpamMailbox(messageMoveEvent)) {
                 LOGGER.debug("Spam event detected");
-                ImmutableList<InputStream> messages = addedToMailboxEvent.getAvailableMessages()
+                ImmutableList<InputStream> messages = messageMoveEvent.getMessages()
                     .values()
                     .stream()
                     .map(Throwing.function(Message::getFullContent))
                     .collect(Guavate.toImmutableList());
-                spamAssassin.learnSpam(messages, addedToMailboxEvent.getMailboxPath().getUser());
+                spamAssassin.learnSpam(messages, messageMoveEvent.getSession().getUser().getUserName());
             }
         }
     }
 
     @VisibleForTesting
-    boolean isEventOnSpamMailbox(MailboxEvent event) {
-        return Role.from(event.getMailboxPath().getName())
-            .filter(role -> role.equals(Role.SPAM))
-            .isPresent();
+    boolean isMessageMovedToSpamMailbox(MessageMoveEvent event) {
+        try {
+            MailboxPath spamMailboxPath = MailboxPath.forUser(event.getSession().getUser().getUserName(), Role.SPAM.getDefaultMailbox());
+            MailboxId spamMailboxId = mapperFactory.getMailboxMapper(event.getSession()).findMailboxByPath(spamMailboxPath).getMailboxId();
```

Comment:
the intent is not very clear, you should probably try to extract things to make the code more obvious

Reply:
This class has changed, I can't link your comment to the actual code; can you help me?

Reply:
isMessageMovedToSpamMailbox calls a lot of methods to just retrieve the Spam mailboxId.